beacon/config.go

```diff
@@ -26,7 +26,7 @@ type Config struct {
 	// Weak coin round duration
 	WeakCoinRoundDuration time.Duration `mapstructure:"beacon-weak-coin-round-duration"`
 	// Ratio of votes for reaching consensus
-	Theta *big.Rat `mapstructure:"beacon-theta"`
+	Theta big.Rat `mapstructure:"beacon-theta"`
```

Comment:
I think we should stick with the pointer to `big.Rat` here. All operations using `big.Rat` take and return pointers, if we use the value type we have to reference and dereference values we receive or pass around.

Reply:
That's true, but on the other hand, having a pointer here makes no logical sense (the value is not optional) and makes it impossible to pass a zero-value of Config to the `AddFlags()` method as it crashes on nil pointer.

IMHO, adding a `*` here and there is a minor inconvenience.

Reply:
Correction - it might not be but only because `types.RatVar` isn't `nil` safe, but it could be:

```go
// RatVar is a wrapper for big.Rat to use it with the pflag package.
type RatVar big.Rat

// String returns a string representation of big.Rat.
func (r *RatVar) String() string {
	if r == nil {
		return ""
	}
	return (*big.Rat)(r).String()
}

// Set sets the value of big.Rat to a string.
func (r *RatVar) Set(s string) error {
	tmp, ok := new(big.Rat).SetString(s)
	if !ok {
		return errors.New("malformed string provided")
	}
	r = tmp
	return nil
}

// Type returns *big.Rat type.
func (r *RatVar) Type() string {
	return "*big.Rat"
}
```

Reply:
> it was also working before the change

It wasn't working before the change for zero-value Config structure as well. It only came up in unit tests because I changed them to pass a zero-value Config (`node_test.go`).